### PR TITLE
Update data models guidelines

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
This is a follow-up of #722 where we set up GraphQL types generation so we don't need to define them in each test data model.

We're updating the docs to reflect those changes in this PR.

Also, we gathered some initial feedback around creating new models and the concept of referenced models does not seem to be very clear so I added a new section trying to explain a bit what they are and how we handle them.

### CODE REVIEW

**Comment by CarlosCortizasCT** [comment]
We can't merge this PR until the previous one (#722) is merged first.

**Review by obulaworld — APPROVED** [review]

**Pull request merged**
